lib/route/nh.c

```diff
@@ -452,6 +552,33 @@ static int nexthop_msg_parser(struct nl_cache_ops *ops, struct sockaddr_nl *who,
 		nexthop->ce_mask |= NH_ATTR_GROUP;
 	}
 
+	/* Parse resilient nexthop group parameters if present */
+	if (tb[NHA_RES_GROUP]) {
+		struct nlattr *rg[NHA_RES_GROUP_MAX + 1];
+
+		if (nla_parse_nested(rg, NHA_RES_GROUP_MAX, tb[NHA_RES_GROUP],
+				     NULL) == 0) {
+			if (rg[NHA_RES_GROUP_BUCKETS]) {
+				nexthop->res_grp_buckets =
+					nla_get_u16(rg[NHA_RES_GROUP_BUCKETS]);
+				nexthop->ce_mask |= NH_ATTR_RES_BUCKETS;
+			}
+			if (rg[NHA_RES_GROUP_IDLE_TIMER]) {
+				nexthop->res_grp_idle_timer = nla_get_u32(
+					rg[NHA_RES_GROUP_IDLE_TIMER]);
+				nexthop->ce_mask |= NH_ATTR_RES_IDLE_TIMER;
+			}
+			if (rg[NHA_RES_GROUP_UNBALANCED_TIMER]) {
+				nexthop->res_grp_unbalanced_timer = nla_get_u32(
+					rg[NHA_RES_GROUP_UNBALANCED_TIMER]);
```

Comment:
you should pass a `policy` argument to `nla_parse_nested()`.

Otherwise, you cannot blindly trust that the provided value is in fact a `nla_get_u32()`.

Theoretically, you could also check each attribute individually. But seems simpler to create a policy.